crates/compiler/alias_analysis/src/lib.rs

```diff
@@ -1589,13 +1621,14 @@ fn layout_spec_help<'a>(
             }
         }
 
-        Boxed(inner_layout) => {
+        Ptr(inner_layout) | Boxed(inner_layout) => {
             let inner_type =
                 layout_spec_help(env, builder, interner, interner.get_repr(inner_layout))?;
             let cell_type = builder.add_heap_cell_type();
```

Comment:
Does morphic distinguish heap/stack alloca values, or are they equivalent?

Reply:
we model the alloca as if it were on the heap here. morphic does not really have a `reference` operator, nor the concept of stack allocations. There are just stack values and everything else.